Parser/action_helpers.c

```diff
@@ -1691,3 +1691,18 @@ _PyPegen_concatenate_strings(Parser *p, asdl_expr_seq *strings,
     assert(current_pos == n_elements);
     return _PyAST_JoinedStr(values, lineno, col_offset, end_lineno, end_col_offset, p->arena);
 }
+
+stmt_ty
+_PyPegen_checked_future_import(Parser *p, identifier module, asdl_alias_seq * names, int level,
```

Comment:
AFAICT, this would be called whenever there is an `from x import y` right? I don't think performances will be impact a lot but is there any possibility for it to reduce import time of modules with lots of imports (it shouldn't be a lot because it's only few string comparisons and modules are cached anyway).

Now, my suggestion: can we use the fact that `__future__` imports are top-levels and thus cannot be nested? (maybe we could use `col_offset`) That could save a streing comparison.

Reply:
I think this is premature optimisation. I ran some benchmarks on the standard library and there is no visible impact

Reply:
My suggestion would have been `if (col_offset == 0 && PyUnicode(...))` (don't remember whether the offset is 0 or 1-based) but let's not overcomplicate the code then!

Reply:
I don't think that will save a lot if you have multiple imports no? If you have hundreds of imports that condition is always true for all of them

Reply:
Also you can do:

```
from __future__ import (
    annotations,
    Barry_as_flufl
)
```

Reply:
Oh, you're right. Nevermind then!